src/lib/builders/tabs/create.ts

```diff
@@ -130,35 +131,44 @@ export function createTabs(props?: CreateTabsProps) {
 				addEventListener(node, 'keydown', (e) => {
 					const tabValue = node.dataset.value;
 
-					const el = e.currentTarget as HTMLElement;
-					const rootEl = el.closest(selector()) as HTMLElement | null;
+					const el = e.currentTarget;
+					if (!isHTMLElement(el)) return;
+					const rootEl = el.closest(selector());
 
-					if (!rootEl || !tabValue) return;
+					if (!isHTMLElement(rootEl) || !tabValue) return;
 
 					const $options = get(options);
 
-					const triggers = Array.from(rootEl.querySelectorAll('[role="tab"]')) as HTMLElement[];
+					const triggers = Array.from(rootEl.querySelectorAll('[role="tab"]'));
 					const enabledTriggers = triggers.filter((el) => !el.hasAttribute('data-disabled'));
```

Comment:
We could possibly filter by htmlElement here and remove the inner if statements. Same for other cases where this happens.